components/NotionList.js

```diff
@@ -0,0 +1,100 @@
+export default function NotionList({
+  $target,
+  initialState,
+  onClick,
+  onAdd,
+  onDelete,
+}) {
+  const $notionList = document.createElement("div");
+  $notionList.className = "notionList";
+
+  $target.appendChild($notionList);
+
+  this.state = initialState;
+
+  this.setState = (nextState) => {
+    this.state = nextState;
+    this.render();
+  };
+
+  this.render = () => {
+    // 하위 항목 표시를 위한 콜백 함수 (재귀 구현)
+    // 제목 길이를 제한하기 위한 유효성 검사 추가!
+    const renderNotion = (id, title, documents) => {
+      if (documents.length > 0) {
+        return `
+        <div>
+          <li class="notion" data-id="${id}">
+            <button class="dropdown"><i class="fa fa-solid fa-caret-up fa-rotate-180 fa-2xs"></i></button>
+            ${title.length > 5 ? title.slice(0, 5) + "..." : title}
+            <div class="btnContainer">
+              <button class="removeBtn" data-id="${id}">❌</button>
+              <button class="addBtn" data-id="${id}">➕</button>
+            </div>
+          </li> 
+          <ul class="children">
+            ${documents
+              .map(
+                ({ id, title, documents }) => `
+              ${renderNotion(id, title, documents)}
```

Comment:
p3;
문자열 반환이라 백킷은 안써도 되겠네요!
```suggestion
              renderNotion(id, title, documents)
```